From 9f03fefb54df7e5e1fbfbd8a528704a7427cd968 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Tue, 13 Jul 2010 15:10:42 +0400
Subject: [PATCH 2/5] No erlang:min(A,B) in R12B-5 and below

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 src/mochiweb.erl |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/src/mochiweb.erl b/src/mochiweb.erl
index 3118028..0e8fedb 100644
--- a/src/mochiweb.erl
+++ b/src/mochiweb.erl
@@ -282,7 +282,10 @@ client_headers(Body, IsLastRequest) ->
 drain_reply(_SockFun, 0, Acc) ->
     Acc;
 drain_reply(SockFun, Length, Acc) ->
-    Sz = erlang:min(Length, 1024),
+    Sz = case (Length < 1024) of
+	true -> Length;
+	_ -> 1024
+    end,
     {ok, B} = SockFun({recv, Sz}),
     drain_reply(SockFun, Length - Sz, <<Acc/bytes, B/bytes>>).
 
-- 
1.7.3.2

